continuum/transforms/custom.py

```diff
@@ -0,0 +1,88 @@
+from typing import Tuple, Union
+import numpy as np
+import torch
+from continuum.datasets import _ContinuumDataset
+
+
+class BackgroundSwap:
+    """Swap input image background with a randomly selected image from bg_images dataset
+
+    :param bg_images: background image dataset, must be normalized.
+    :param bg_label: label class from background image set
+    :param input_dim: input dimension of transform, excluding channels
+    """
+
+    def __init__(self, bg_images: _ContinuumDataset, input_dim: Tuple[int, int] = (28, 28),
+                 bg_label: int = None,
+                 normalize_bg: bool = True):
+        self.bg_label = bg_label
+        self.normalize_bg = normalize_bg
+        self.input_dim = input_dim
+
+        full = bg_images.get_data()
+
+        if bg_label is not None:
+            self.bg_images = full[0][np.where(full[1] == bg_label)]
+        else:
+            self.bg_images = full[0]
+
+    def _randcrop(self, img: Union[np.ndarray, torch.Tensor]) -> Union[np.ndarray, torch.Tensor]:
+        """Crop input image to self.input_dim shape
+        :param img: input image
+        """
+
+        crop_height = img.shape[0] >= self.input_dim[0]
+        crop_width = img.shape[1] >= self.input_dim[1]
+
+        x_crop = np.random.randint(0, img.shape[0] - self.input_dim[0])
+        y_crop = np.random.randint(0, img.shape[1] - self.input_dim[1])
+
+        if crop_width and crop_height:
+            return img[x_crop:x_crop + self.input_dim[0], y_crop: y_crop + self.input_dim[1], :]
+
+        if crop_width:
+            return img[:, y_crop: y_crop + self.input_dim[1], :]
+
+        if crop_height:
+            return img[x_crop:x_crop + self.input_dim[0], :, :]
+
+        raise Exception("Background image is smaller than foreground image")
```

Comment:
I don't reach this exception because I first have the `ValueError: low >= high` exception in the `randint` method as `img.shape[0] - self.input_dim[0]` will clearly be negative and thus lower than the low value (0).

Reply:
Btw, could we resize the bg image in that case? Using `scipy.misc.imresize`

Reply:
I think we could actually just resize the image and remove def _randcrop() completely